microsoft/testsuites/kselftest/kselftest.py

```diff
@@ -0,0 +1,285 @@
+import os
+import re
+from dataclasses import dataclass
+from pathlib import PurePath, PurePosixPath
+from typing import Any, Dict, List
+
+from assertpy import assert_that
+
+from lisa import Environment, notifier
+from lisa.base_tools.uname import Uname
+from lisa.executable import Tool
+from lisa.messages import SubTestMessage, TestStatus, create_test_result_message
+from lisa.node import Node
+from lisa.operating_system import CBLMariner
+from lisa.testsuite import TestResult
+from lisa.tools import Cp, Git, Ls, Make, RemoteCopy, Tar
+from lisa.tools.whoami import Whoami
+from lisa.util import (
+    LisaException,
+    SkippedException,
+    UnsupportedDistroException,
+    find_groups_in_lines,
+)
+
+
+@dataclass
+class KselftestResult:
+    name: str = ""
+    status: TestStatus = TestStatus.QUEUED
+    exit_value: int = 0
+
+
+class Kselftest(Tool):
+    _MARINER_KERNEL_SRC_REPO = "https://github.com/microsoft/CBL-Mariner-Linux-Kernel"
+    _KERNEL_REPO_NAME = "CBL-Mariner-Linux-Kernel"
+    _KSELFTEST_TAR_PATH = (
+        "/build/kselftest/kselftest_install/kselftest-packages/kselftest.tar.gz"
+    )
+
+    # kselftest result log has "ok" and "not ok" prefixes, use regex to filter them
+    # [{'status': 'not ok', 'name': 'step_after_suspend_test', 'info': '# exit=126'}]
+    _RESULT_KSELFTEST_OK_REGEX = re.compile(
+        r"^(?P<status>(not ok|ok))\s+\d+\s+selftests:\s+\S+:\s+(?P<name>\S+)\s*(?P<info>.*)$"  # noqa: E501
```

Comment:
The regexp can be more accurate to just capture the values, which are needed, so it doesn't need to split string again in code.

Reply:
> The regexp can be more accurate to just capture the values, which are needed, so it doesn't need to split string again in code.

I tried to do this but we will still have to split the info string further as it can have 4 different possibilities:

not ok 1 selftests: vm: run_vmtests.sh # TIMEOUT 45 seconds
ok 2 selftests: cgroup: test_kmem # SKIP
not ok 4 selftests: clone3: clone3_cap_checkpoint_restore # exit=1
ok 3 selftests: clone3: clone3_set_tid

Reply:
> > The regexp can be more accurate to just capture the values, which are needed, so it doesn't need to split string again in code.
> 
> I tried to do this but we will still have to split the info string further as it can have 4 different possibilities:
> 
> not ok 1 selftests: vm: run_vmtests.sh # TIMEOUT 45 seconds ok 2 selftests: cgroup: test_kmem # SKIP not ok 4 selftests: clone3: clone3_cap_checkpoint_restore # exit=1 ok 3 selftests: clone3: clone3_set_tid

Thanks for helping fix this, I have updated parsing logic.